extensions/jackson/runtime/src/main/java/io/quarkus/jackson/runtime/ObjectMapperProducer.java

```diff
@@ -63,6 +64,9 @@ public ObjectMapper objectMapper(@All List<ObjectMapperCustomizer> customizers,
         for (ObjectMapperCustomizer customizer : sortedCustomizers) {
             customizer.customize(objectMapper);
         }
+        if (true) {
+            objectMapper.getFactory().setBufferRecyclerPool(BufferRecyclerPool.LockFreePool.shared());
```

Comment:
Here I'm just demonstrating how to configure a different pool implementation using the new Jackson API. The specific pool that I'm using is one of the implementations now offered out of the box by Jackson. This pool works reasonably well also with virtual threads, but it's pretty simple also because one of the constraint understandably required by Jackson maintainers was to not add any dependency to `jackson-core`. I believe that we could develop our own pool implementation, for instance based on some `JCTools` highly concurrent data structures, and if we find that it performs satisfyingly in all conditions (with both native and virtual threads) we may also decide to always use it regardless.